![Screenshot_2020-11-12 PowerPoint Presentation - TVM pdf](https://user-images.githubusercontent.com/33932398/98984067-73535180-2500-11eb-81dd-f7e8acb430f3.png)

![Screenshot_2020-11-12 PowerPoint Presentation - TVM pdf(1)](https://user-images.githubusercontent.com/33932398/98984150-92ea7a00-2500-11eb-84c0-7c17d4872580.png)

![Screenshot_2020-11-12 PowerPoint Presentation - TVM pdf(2)](https://user-images.githubusercontent.com/33932398/98984222-ab5a9480-2500-11eb-8bae-1f57e2108f58.png)


# Coding Exercises 

--------------------------------------------------------------------------------------------------------------

## Exercise 1: Using Python as a Calculator for TVM Problems

Use __variables__ where appropriate! __Input__ values and your __final results__ shall be stored in variables. 

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [76]:
present_value = 2000
interest_rate = 1.04
num_of_periods = 7

future_value = present_value*interest_rate**num_of_periods
print("Future Value:",round(future_value,2),"USD")

Future Value: 2631.86 USD


(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a. 

In [77]:
future_value = 20000
interest_rate = 1.035
num_of_periods= 4
present_value = future_value/(interest_rate**num_of_periods)

In [78]:
print("Present Value:",round(present_value,2),"USD")

Present Value: 17428.84 USD


(result: 17,428.84 USD)

![Screenshot_2020-11-12 PowerPoint Presentation - Interest_Rates pdf](https://user-images.githubusercontent.com/33932398/98985805-d8a84200-2502-11eb-9acd-fe80fa0bd68a.png)

![Screenshot_2020-11-12 PowerPoint Presentation - Interest_Rates pdf(1)](https://user-images.githubusercontent.com/33932398/98986315-81ef3800-2503-11eb-88fa-882743a86faa.png)

![Screenshot_2020-11-12 PowerPoint Presentation - Interest_Rates pdf(2)](https://user-images.githubusercontent.com/33932398/98986546-d09cd200-2503-11eb-93dd-91cc8cae6c87.png)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [83]:
present_value = 10000
future_value  = 15000
num_of_periods= 10

interest_rate = ((future_value/present_value)** (1/num_of_periods) - 1) * 100

In [84]:
print("Interest Rate",round(interest_rate,2),"%")

Interest Rate 4.14 %


(result: 4.14%)

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__! 

In [85]:
old_values = 976
current_value = 1053
price_return = ((current_value - old_values)/old_values) * 100

In [100]:
print("Price Return:",round(price_return,2),"%")

Price Return: 7.89 %


(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [101]:
dividend_payment = 40
old_values = 976
current_value = 1053
dividend_yield = (dividend_payment/old_values) * 100
total_returns  = ((current_value + dividend_payment)/old_values-1) * 100

In [103]:
print("Dividend Yield:",round(dividend_yield,2),"% |Total Return:",round(total_returns,2),"%")


Dividend Yield: 4.1 % |Total Return: 11.99 %


(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [104]:
print("Price Return:",round(price_return,2),"% |Dividend Yield:",round(dividend_yield,2),"% |Total Return:",round(total_returns,2),"%")


Price Return: 7.89 % |Dividend Yield: 4.1 % |Total Return: 11.99 %


---------------------------------------------------------------------------------------------------------------------